bql/semantic/expression.go

```diff
@@ -87,6 +87,80 @@ type evaluationNode struct {
 	rB string
 }
 
+type evaluationLiteralNode struct {
+	op OP
+
+	literalOnLeft bool
+
+	lS string
+	rS string
+}
+
+func (e *evaluationLiteralNode) Evaluate(r table.Row) (bool, error) {
+	// Binary evaluation
+	getLeft := func() (*table.Cell, error) {
+		var (
+			eL *table.Cell
+			ok bool
+		)
+		eL, ok = r[e.lS]
+		if !ok {
+			return nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.lS, r)
+		}
+		return eL, nil
+	}
+	getRight := func() (*table.Cell, error) {
+		var (
+			eR *table.Cell
+			ok bool
+		)
+		eR, ok = r[e.rS]
+		if !ok {
+			return nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.rS, r)
+		}
+		return eR, nil
+	}
+
+	cs := func(c *table.Cell) string {
+		if c.L != nil {
+			return strings.TrimSpace(c.L.ToComparableString())
+		}
+		return strings.TrimSpace(c.String())
+	}
+
+	leftCell, lerr := getLeft()
+	rightCell, rerr := getRight()
+	var (
+		csEL, csER string
+	)
+
+	if lerr != nil && rerr != nil {
+		return false, fmt.Errorf("comparison operations require either the left or right operands to be a binding but neither had an existing row")
+	}
+
+	if lerr == nil {
+		csEL = cs(leftCell)
+	} else {
```

Comment:
Given the code above, else will not be called. Also, see comment above about early return.

Reply:
the error actually signals us that that is a literal and not a binding, so, in the if above we check if there are two literals, if that is the case we bail out.

below, in the else we assign csEL to the literal value when there is an error.